packages/headless/src/api/service/case-assist/case-assist-params.ts

```diff
@@ -0,0 +1,85 @@
+import {HTTPContentType, HttpMethods} from '../../platform-client';
+import {BaseParam} from '../../platform-service-params';
+
+export interface CaseAssistIdParam {
+  caseAssistId: string;
+}
+
+export interface FieldsParam {
+  fields: Record<string, string>;
+}
+
+export type CaseAssistParam = BaseParam & CaseAssistIdParam;
+
+/**
+ * Builds a base request for calling the Case Assist API.
+ *
+ * @param req - The Case Assist request parameters.
+ * @param method - The HTTP method used to issue the request.
+ * @param contentType - The request content type.
+ * @param path - The request path, relative to the Case Assist path.
+ * @param queryStringArguments - The arguments to pass in the query string.
+ * @returns The built request information.
+ */
+export const baseCaseAssistRequest = (
+  req: CaseAssistParam,
+  method: HttpMethods,
+  contentType: HTTPContentType,
+  path: string,
+  queryStringArguments: Record<string, string> = {}
+) => {
+  validateCaseAssistRequestParams(req);
+
+  const baseUrl = `${req.url}/rest/organizations/${req.organizationId}/caseassists/${req.caseAssistId}${path}`;
```

Comment:
Remember that sanitization done upstream with Bueno (at an action level) will be important to ensure those params valid. 

eg: ensuring a regex match on "a valid case assist id" when it is set through an action.

Reply:
:thinking: I thought I could simply pass the `caseAssistId` in the engine configuration, and then use it. I now realize that I forgot to update `HeadlessEngine.validateConfiguration()`. I think I'll also have to dispatch an `updateCaseAssistConfiguration` action, validate the ID again, and update the related state interfaces.

Reply:
What you propose has some impacts on the way engines are loaded in quantic. But, I'll come up with something.

Reply:
As mentioned in another comment, I'll remove the engine integration part until we figure out how to properly connect multiple API clients with the engine.